lib/THC/THCTensorSort.cu

```diff
@@ -28,303 +22,242 @@ unsigned long nextHighestPowerOf2(unsigned long n) {
-        sortedInfo, indicesInfo, inputInfo,                             \
-        slices, (TYPE) sliceSize, (TYPE) sliceStride,                   \
-        (TYPE) outSize, (TYPE) outStride,                               \
+        keyInfo,                                                        \
+        keySlices,                                                      \
+        (TYPE) keySliceSize,                                            \
+        (TYPE) keyInfo.strides[collapseKeyDim],                         \
+        valueInfo,                                                      \
+        (TYPE) valueInfo.strides[collapseValueDim],                     \
         LTComp<float>());                                               \
   }
 
-#define HANDLE_SORT_CASE(TYPE, A)               \
-  {                                             \
-    switch (ceilPowerOf2) {                     \
-      case 2048:                                \
-      HANDLE_CASE(TYPE, A, 2048);               \
-      break;                                    \
-      case 1024:                                \
-      HANDLE_CASE(TYPE, A, 1024);               \
-      break;                                    \
-      case 512:                                 \
-      HANDLE_CASE(TYPE, A, 512);                \
-      break;                                    \
-      case 256:                                 \
-      HANDLE_CASE(TYPE, A, 256);                \
-      break;                                    \
-      case 128:                                 \
-      HANDLE_CASE(TYPE, A, 128);                \
-      break;                                    \
-      case 64:                                  \
-      HANDLE_CASE(TYPE, A, 64);                 \
-      break;                                    \
-      case 32:                                  \
-      HANDLE_CASE(TYPE, A, 32);                 \
-      break;                                    \
-      case 16:                                  \
-      HANDLE_CASE(TYPE, A, 16);                 \
-      break;                                    \
-      case 8:                                   \
-      HANDLE_CASE(TYPE, A, 8);                  \
-      break;                                    \
-      case 4:                                   \
-      HANDLE_CASE(TYPE, A, 4);                  \
-      break;                                    \
-      case 2:                                   \
-      HANDLE_CASE(TYPE, A, 2);                  \
-      break;                                    \
-      case 1:                                   \
-      HANDLE_CASE(TYPE, A, 1);                  \
-      break;                                    \
-      default:                                  \
-      assert(false);                            \
-    }                                           \
+#define HANDLE_SORT_CASE(TYPE, A)                       \
```

Comment:
if it were C++11, constexpr and some template magic can do this macro expansion crap, but as a series of successive `if` statements rather than a switch.